### PR TITLE
feature/generator-for-database-cleaner

### PR DESCRIPTION
**Pull request by TheZero0-ctrl**
Tasks Done: 
- Added generator for database_cleaner for RSpec

### CODE REVIEW

**Comment by TheZero0-ctrl** [comment]
@abhaynikam Could you please review this PR at your convenience and let me know if any changes are necessary? Thank you for your time.

File: lib/generators/boring/database_cleaner/active_record/rspec/install/install_generator.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      module Rspec
+        class InstallGenerator < Rails::Generators::Base
+          desc 'Adds database_cleaner-active_record gem to the application for cleaning databases during tests'
+          source_root File.expand_path("templates", __dir__)
+
+          TEST_FRAMEWORK_GEM = 'rspec-rails'
+
+          class_option :skip_js_configs, type: :boolean,
+                                               aliases: "-sj",
+                                               desc: "Skip JacaScript dependent configurations"
```
Comment (abhaynikam) [inline lib/generators/boring/database_cleaner/active_record/rspec/install/install_generator.rb:15]:
```suggestion
                                               desc: "Skip Javascript configurations"
```

File: lib/generators/boring/database_cleaner/active_record/rspec/install/install_generator.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+          def verify_presence_of_rspec_gem
+            gem_file_content_array = File.readlines("Gemfile")
+            rspec_is_installed = gem_file_content_array.any? { |line| line.include?(TEST_FRAMEWORK_GEM) }
+
+            return if rspec_is_installed
+
+            say "We couldn't find #{TEST_FRAMEWORK_GEM} gem. Please configure #{TEST_FRAMEWORK_GEM} and run the generator again!", :red
```
Comment (abhaynikam) [inline lib/generators/boring/database_cleaner/active_record/rspec/install/install_generator.rb:23]: can we suggest running rspec generator here?

File: lib/generators/boring/database_cleaner/active_record/rspec/install/install_generator.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            say "We couldn't find spec/rails_helper.rb. Please configure RSpec and run the generator again!", :red
+
+            abort
+          end
+
+          def add_database_cleaner_active_record_gem
+            say 'Adding database_cleaner-active_record gem', :green
+
+            Bundler.with_unbundled_env do
+              run "bundle add database_cleaner-active_record --group='test'"
```
Comment (abhaynikam) [inline lib/generators/boring/database_cleaner/active_record/rspec/install/install_generator.rb:40]: using `bundle add --group` adds new line to the gem file and I feel most people(including me) would be annoyed if all test group gems are not clubbed under one block.

I would recommend using `insert_into_file` with regex into the test group. We have already done that in some of the generator so should be easy to copy here. 

WDYT?

**Review by abhaynikam — APPROVED** [review]
Left some comments - otherwise looks good to me.

**Comment by TheZero0-ctrl** [comment]
It seems database cleaner is not needed for rails 7+
can you confirm if we still need this generator?
@coolprobn

**Comment by coolprobn** [comment]
> It seems database cleaner is not needed for rails 7+ can you confirm if we still need this generator? @coolprobn

Yep I agree, we can close this one actually.

We only realized it after creating the PR but forgot to close it.

@TheZero0-ctrl

**Pull request closed without merging**